### PR TITLE
Overloaded varargs

### PR DESCRIPTION
**Pull request by andvazqu**
See https://github.com/ninia/jep/issues/451

### CODE REVIEW

**Comment by ndjensen** [comment]
This looks good to me.  Is it possible you could add a test to the end of test_call.py where the test fails without your fix and then passes with your fix?

File: src/main/c/Objects/pyjmethod.c [hunk 1]
```diff
@@ -186,6 +187,11 @@ int PyJMethod_CheckArguments(PyJMethodObject* method, JNIEnv *env,
         paramTypeId = get_jtype(env, paramType);
 
         match = pyarg_matches_jtype(env, param, paramType, paramTypeId);
+        if (match == 0 && varargs && paramTypeId == JARRAY_ID) {
```
Comment (bsteffensmeier) [inline src/main/c/Objects/pyjmethod.c:190]: This if statement needs to be limited to just the varargs parameter. If there is a method that takes an array and varargs then this will match the component type to the array.

For example this java method
```java
public String varargs(String[] arr, Object... varargs);
```
Would match this python call.
```python
obj.varargs("String")
```
Even though the call will fail.

**Review by bsteffensmeier — CHANGES_REQUESTED** [review]
Thanks for putting this together. I have identified one edge case in an inline comment that should be fixed before this can be merged.

I agree unit tests would be nice, I  recommend test_overload instead of test_call.

**Comment by bsteffensmeier** [comment]
The actual change in the jep code looks good to me. One of the unit tests is failing in AppVeyor in some environments for your test:

> Traceback (most recent call last):
>   File "C:\projects\jep\src\test\python\test_overload.py", line 86, in test_varargs
>     self.assertEqual(TestOverload.varargs(0), 'int...args')
> AssertionError: 'int i, String...args' != 'int...args'

I think that call is ambiguous and don't neither method seems more correct to me so in my opinion the test should pass if either method is used.

**Review by bsteffensmeier — APPROVED** [review]

**Comment by bsteffensmeier** [comment]
I'll be merging this in to the dev_4.2 branch, which I have not created yet so this is waiting on me now. Thank you very much for the contribution.
